tests/e2e/dsc_cfmap_deletion_test.go

```diff
@@ -26,99 +23,81 @@ func cfgMapDeletionTestSuite(t *testing.T) {
-		})
-
-		t.Run("owned namespaces should be deleted", func(t *testing.T) {
-			err = testCtx.testOwnedNamespacesDeletion()
+		t.Run("owned namespaces should be not deleted", func(t *testing.T) {
+			err = testCtx.testOwnedNamespacesAllExist(0)
 			require.NoError(t, err, "Error while deleting owned namespaces")
 		})
-
-		t.Run("dsci should be deleted", func(t *testing.T) {
-			err = testCtx.testDSCIDeletion()
-			require.NoError(t, err, "failed deleting DSCI")
-		})
-
-		t.Run("applications resources should be deleted", func(t *testing.T) {
-			err = testCtx.testAllApplicationDeletion()
-			require.NoError(t, err, "Error to delete component")
-		})
 	})
 }
 
-func (tc *testContext) testDSCIDeletion() error {
-	dsciInstances := &dsci.DSCInitializationList{}
-	if err := tc.customClient.List(context.TODO(), dsciInstances); err != nil {
-		return fmt.Errorf("failed while listing DSCIs: %w", err)
-	}
-
-	if len(dsciInstances.Items) != 0 {
-		return fmt.Errorf("expected DSCI removal, but got %v", dsciInstances)
-	}
-
-	return nil
-}
-
-func (tc *testContext) testDSCDeletionUsingConfigMap() error {
+func (tc *testContext) testDSCDeletionUsingConfigMap(enableDeletion string) error {
 	dscLookupKey := types.NamespacedName{Name: tc.testDsc.Name}
 	expectedDSC := &dsc.DataScienceCluster{}
 
-	if err := createDeletionConfigMap(tc); err != nil {
+	if err := createDeletionConfigMap(tc, enableDeletion); err != nil {
 		return err
 	}
 
 	err := tc.customClient.Get(tc.ctx, dscLookupKey, expectedDSC)
-	if err == nil {
-		dscerr := tc.customClient.Delete(tc.ctx, expectedDSC, &client.DeleteOptions{})
-		if dscerr != nil {
-			return fmt.Errorf("error deleting DSC instance %s: %w", expectedDSC.Name, dscerr)
-		}
-	} else if !k8serrors.IsNotFound(err) {
-		if err != nil {
-			return fmt.Errorf("error getting DSC instance :%w", err)
+	// should have DSC instance
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return fmt.Errorf("should have DSC instance in cluster:%w", err)
 		}
+		return fmt.Errorf("error getting DSC instance :%w", err)
 	}
 
 	return nil
 }
+func (tc *testContext) testOwnedNamespacesAllExist(count int) error {
+	namespaces, err := tc.kubeClient.CoreV1().Namespaces().List(tc.ctx, metav1.ListOptions{
+		LabelSelector: labels.ODH.OwnedNamespace,
+	})
 
-func (tc *testContext) testOwnedNamespacesDeletion() error {
-	if err := wait.PollUntilContextTimeout(tc.ctx, tc.resourceRetryInterval, tc.resourceCreationTimeout, false, func(ctx context.Context) (bool, error) {
-		namespaces, err := tc.kubeClient.CoreV1().Namespaces().List(ctx, metav1.ListOptions{
-			LabelSelector: labels.ODH.OwnedNamespace,
-		})
-
-		return len(namespaces.Items) == 0, err
-	}); err != nil {
-		return fmt.Errorf("failed waiting for all owned namespaces to be deleted: %w", err)
+	if err != nil {
+		return fmt.Errorf("failed getting owned namespaces %w", err)
+	}
+	if len(namespaces.Items) == count {
+		return fmt.Errorf("all namespaces are gone")
```

Comment:
Strange message, if count is not 0?

Reply:
this is to check if we have any namespaces existing in the cluster, 
if count is 0, as all namespace with matched labels are gone, should is triggered by the ODH deletion, in the case of a configmap with "true" on its label

Reply:
With the removed parameter and comparation with 0 it's clear